src/openms/source/FILTERING/DATAREDUCTION/SplineInterpolatedPeaks.cpp

```diff
@@ -43,226 +43,221 @@ using namespace std;
       next_intensity_zero = (intensity[i + 1] == 0);
       if (!last_intensity_zero || !current_intensity_zero || !next_intensity_zero)
       {
-        mz_slim1.push_back(mz[i]);
+        pos_slim1.push_back(pos[i]);
         intensity_slim1.push_back(intensity[i]);
       }
     }
-    if (intensity[mz.size() - 1] != 0 || intensity[mz.size() - 2] != 0)
+    if (intensity[pos.size() - 1] != 0 || intensity[pos.size() - 2] != 0)
     {
-      mz_slim1.push_back(mz[mz.size() - 1]);
-      intensity_slim1.push_back(intensity[mz.size() - 1]);
+      pos_slim1.push_back(pos[pos.size() - 1]);
+      intensity_slim1.push_back(intensity[pos.size() - 1]);
     }
 
     // remove Thermo bug zeros
     // (In some Thermo data appear odd zero intensity data points. Normal data points are sometimes quickly followed by a zero.
     // These zeros are clearly not part of the profile, but bugs. The following code snippet removes them. A datapoint is
     // "quickly followed" by a second one, if the m/z step is shorter than scaling_Thermo_bug times the previous m/z step.)
-    std::vector<double> mz_slim2; // slimmer vector after removal of Thermo bugs from mz_slim1
+    std::vector<double> pos_slim2; // slimmer vector after removal of Thermo bugs from pos_slim1
     std::vector<double> intensity_slim2; // slimmer vector after removal of Thermo bugs from intensity_slim1
     const double scaling_Thermo_bug = 1.0 / 50.0; // scaling factor for Thermo bug
-    mz_slim2.reserve(mz_slim1.size());
+    pos_slim2.reserve(pos_slim1.size());
     intensity_slim2.reserve(intensity_slim1.size());
-    mz_slim2.push_back(mz_slim1[0]);
-    mz_slim2.push_back(mz_slim1[1]);
+    pos_slim2.push_back(pos_slim1[0]);
+    pos_slim2.push_back(pos_slim1[1]);
     intensity_slim2.push_back(intensity_slim1[0]);
     intensity_slim2.push_back(intensity_slim1[1]);
-    for (size_t i = 2; i < mz_slim1.size(); ++i)
+    for (size_t i = 2; i < pos_slim1.size(); ++i)
     {
       if (intensity_slim1[i] == 0)
       {
-        if ((mz_slim1[i] - mz_slim1[i - 1]) < (mz_slim1[i - 1] - mz_slim1[i - 2]) * scaling_Thermo_bug)
+        if ((pos_slim1[i] - pos_slim1[i - 1]) < (pos_slim1[i - 1] - pos_slim1[i - 2]) * scaling_Thermo_bug)
         {
           continue;
         }
       }
-      mz_slim2.push_back(mz_slim1[i]);
+      pos_slim2.push_back(pos_slim1[i]);
       intensity_slim2.push_back(intensity_slim1[i]);
     }
 
     // subdivide spectrum into packages
     std::vector<bool> start_package;
     start_package.push_back(true);
     start_package.push_back(false);
-    for (size_t i = 2; i < mz_slim2.size(); ++i)
+    for (size_t i = 2; i < pos_slim2.size(); ++i)
     {
-      start_package.push_back((mz_slim2[i] - mz_slim2[i - 1]) / (mz_slim2[i - 1] - mz_slim2[i - 2]) > new_package);
+      start_package.push_back((pos_slim2[i] - pos_slim2[i - 1]) / (pos_slim2[i - 1] - pos_slim2[i - 2]) > new_package);
     }
 
     // fill the packages
```

Comment:
have you considered using a different name than package? or at least adding more documentation what a package is